src/Draco.Compiler/Api/CodeCompletion/CompletionService.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Draco.Compiler.Api.Semantics;
+using Draco.Compiler.Api.Syntax;
+
+namespace Draco.Compiler.Api.CodeCompletion;
+
+/// <summary>
+/// Allows to get <see cref="CompletionItem"/>s from multiple <see cref="CompletionProvider"/>s.
```

Comment:
That's not a good description. The service stores registered providers, determines which providers get to do completion, and aggregates the results of them